pkg/gc/config.go

```diff
@@ -62,12 +85,28 @@ func NewConfigFromConfigMapFunc(ctx context.Context) func(configMap *corev1.Conf
 			cm.AsDuration("stale-revision-create-delay", &c.StaleRevisionCreateDelay),
 			cm.AsDuration("stale-revision-timeout", &c.StaleRevisionTimeout),
 			cm.AsDuration("stale-revision-lastpinned-debounce", &c.StaleRevisionLastpinnedDebounce),
-
 			cm.AsInt64("stale-revision-minimum-generations", &c.StaleRevisionMinimumGenerations),
+
+			cm.AsDuration("gc-retain-since-create-time", &c.GCRetainSinceCreateTime),
+			cm.AsDuration("gc-retain-since-last-active-time", &c.GCRetainSinceLastActiveTime),
+			cm.AsInt64("gc-min-stale-revisions", &c.GCMinStaleRevisions),
+			cm.AsInt64("gc-max-stale-revisions", &c.GCMaxStaleRevisions),
 		); err != nil {
 			return nil, fmt.Errorf("failed to parse data: %w", err)
 		}
 
+		if c.GCMaxStaleRevisions != -1 && c.GCMinStaleRevisions > c.GCMaxStaleRevisions {
+			return nil, fmt.Errorf(
+				"gc-min-stale-revisions must be <= gc-max-stale-revisions was %d and %d",
+				c.GCMinStaleRevisions, c.GCMaxStaleRevisions)
+		}
+		if c.GCMinStaleRevisions < 0 {
+			return nil, fmt.Errorf("gc-min-stale-revisions must be non-negative, was: %d", c.GCMinStaleRevisions)
+		}
+		if c.GCMaxStaleRevisions < -1 {
```

Comment:
What does GCMaxStaleRevisions==0 mean? Immediately reap the revision as soon as it becomes non-routable?

Reply:
Correct.

Reply:
added on to the yaml comment